helper/terraformtype/helper/schema/type_schema.go

```diff
@@ -80,7 +85,16 @@ func NewSchemaInfo(cl *ast.CompositeLit, info *types.Info) *SchemaInfo {
 	}
 
 	if kvExpr := result.Fields[SchemaFieldDefault]; kvExpr != nil && astutils.ExprValue(kvExpr.Value) != nil {
-		result.Schema.Default = func() (interface{}, error) { return nil, nil }
+		switch result.Schema.Type {
+		case tfschema.TypeBool:
+			result.Schema.Default = *astutils.ExprBoolValue(kvExpr.Value)
+		case tfschema.TypeInt:
+			result.Schema.Default = *astutils.ExprIntValue(kvExpr.Value)
+		case tfschema.TypeString:
+			result.Schema.Default = *astutils.ExprStringValue(kvExpr.Value)
+		default:
+			result.Schema.Default = func() (interface{}, error) { return nil, nil }
+		}
 	}
```

Comment:
I'm a little worried here about the tight coupling of `Type` to `Default` -- it seems a misconfigured schema could cause a panic.

Reply:
You're right, I didn't come up with this point :smile: So how about adding some error handling here?

Reply:
Adding error checking would be acceptable. 👍